src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -216,6 +218,31 @@ private void DetermineSerializationCapabilities()
             }
         }
 
+        /// <summary>
+        /// Return the JsonPropertyInfo for the TValue in IDictionary{string, TValue} when deserializing.
+        /// </summary>
+        /// <remarks>
+        /// This should not be called during warm-up (initial creation of JsonPropertyInfos) to avoid recursive behavior
+        /// which could result in a StackOverflowException.
+        /// </remarks>
+        public JsonPropertyInfo DictionaryValuePropertyPolicy
+        {
+            get
+            {
+                if (_dictionaryValuePropertyPolicy == null)
+                {
+                    Debug.Assert(ClassType == ClassType.Dictionary || ClassType == ClassType.IDictionaryConstructible);
+
+                    Type dictionaryValueType = ElementType;
```

Comment:
Nit: should the debug assert come first?

Reply:
Sure. Moved it.